aliyun-net-sdk-core/Endpoints/EndpointsProvider.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+using Aliyun.Acs.Core.Auth;
+using Aliyun.Acs.Core.Exceptions;
+using Aliyun.Acs.Core.Profile;
+using Aliyun.Acs.Core.Regions;
+using Aliyun.Acs.Core.Regions.Location.Model;
+using Aliyun.Acs.Core.Utils;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Aliyun.Acs.Core.Endpoints
+{
+    public class EndpointsDataProvider
+    {
+        private const string BUNDLED_ENDPOINTS_RESOURCE_PATH = "endpoints.json";
+
+        private static Dictionary<string, Endpoint> endpoints = new Dictionary<string, Endpoint>();
```

Comment:
Dictionary<TKey,TValue> should be ConcurentDictionary<TKey,TValue> to avoid parallel add value